src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -125,14 +125,15 @@
                       (try
                         ;; Try to get a connection to make sure the db is ready
                         (.close (.getConnection datasource))
-                        (when-let [{:keys [migration-db migration-dir]} migration-opts]
+                        (when-let [{:keys [migration-db migration-dirs]} migration-opts]
                           (log/debug (trs "{0} - Starting database migration" (.getPoolName datasource)))
                           ;; If we're a replica then pglogical will be
                           ;; replicating our migrations for us, so we poll until
                           ;; the migrations have been replicated
                           (if (= (:replication-mode migration-opts) :replica)
-                            (wait-for-migrations migration-db migration-dir)
-                            (migration/migrate migration-db migration-dir)))
+                            (doseq [single-dir migration-dirs]
+                              (wait-for-migrations migration-db single-dir)
+                              (migration/migrate migration-db single-dir))))
```

Comment:
```suggestion
                            (let [migration-dirs (get migration-opts :migration-dirs [migration-dir])
                               (doseq [single-dir migration-dirs]
                                 (wait-for-migrations migration-db single-dir)
                                 (migration/migrate migration-db single-dir)))))
```

Reply:
This change will prefer migration-dir and turn it into a vector if "migration-dirs" is not present.  That should preserve backwards compatibility.

Reply:
Actually I don't think this is quite right.  We only want to do the wait-for-migrations when we are in replica mode and otherwise run the migrations.  This will need some additional restructuring beyond what I gave suggested to break up the two branches.